src/icrc1_auction_api_new.mo

```diff
@@ -236,14 +236,20 @@ actor class Icrc1AuctionAPI(quoteLedger_ : ?Principal, adminPrincipal_ : ?Princi
     let assetInfo = Vec.get(assets, assetId);
     let res = await* assetInfo.handler.depositFromAllowance(caller, args.from, args.amount, args.expected_fee);
     switch (res) {
-      case (#ok(credited, txid)) {
-        assert assetInfo.handler.debitUser(caller, credited);
-        ignore a.appendCredit(caller, assetId, credited);
-        #Ok({
-          credit_inc = credited;
-          txid = txid;
-          credit = a.getCredit(caller, assetId).available;
-        });
+      case (#ok(_, txid)) {
+        let userCredit = assetInfo.handler.userCredit(caller);
+        if (userCredit > 0) {
+          let credited = Int.abs(userCredit);
+          assert assetInfo.handler.debitUser(caller, credited);
+          ignore a.appendCredit(caller, assetId, credited);
+          #Ok({
+            credit_inc = credited;
```

Comment:
Here credit_inc = credit_inc
Like we do in icrc84_notify

Reply:
Or, alternatively, we can change icrc84_notify to do it like you solved it here. The two functions should just be consistent.

Reply:
Got it, thanks. Fixed now